internal/jimm/controller.go

```diff
@@ -505,11 +514,28 @@ func (j *JIMM) ImportModel(ctx context.Context, user *openfga.User, controllerNa
 	model.CloudRegionID = cr.ID
 	model.CloudRegion = cr
 
-	err = j.Database.AddModel(ctx, &model)
-	if err != nil {
-		if errors.ErrorCode(err) == errors.CodeAlreadyExists {
-			return errors.E(op, err, "model already exists")
+	err = j.Database.Transaction(func(d *db.Database) error {
+		err := j.Database.AddModel(ctx, &model)
+		if err != nil {
+			if errors.ErrorCode(err) == errors.CodeAlreadyExists {
+				return errors.E(op, err, "model already exists")
+			}
+			return errors.E(op, err)
+		}
+		for _, offer := range offersToAdd {
+			var dbOffer dbmodel.ApplicationOffer
```

Comment:
while this will add offers to JIMM's database, it will not grant anybody access to imported offers. At the very least, the user importing the model should be granted access.

this will require a bit of thought: each offer comes with a set of users that have access to the offer. If those are external users (those that JIMM cares about), they should be granted access to the offer when it is imported. But i guess in most cases these users would be local users - access to the offer will remain granted on the controller, but JIMM will not know/care about those rights.  perhaps we should simply print this info and let the user importing the model handle it. e.g. we imported this model and these app offer and these are the users that have access to the app offer - handle it!

Reply:
True, I think we need to create an OpenFGA relation between the offers and the model. That way the model owner will automatically get access. Then they can grant access to other users. The users on the controller should never (at least with the new JIMM) be external users, because nowadays those are never stored on Juju's end (only in JIMM) so I think we can safely discard the users coming from the controller.

But yes printing a message would be helpful.

Reply:
+1, relation makes sense to me.

Reply:
Can do in a follow up PR @alesstimec